### PR TITLE
prometheus-operator: Build admission webhook image

### PR DESCRIPTION
**Pull request by philipgough**

### CODE REVIEW

File: ci-operator/config/openshift/prometheus-operator/openshift-prometheus-operator-master.yaml [hunk 1]
```diff
@@ -25,6 +25,13 @@ images:
       as:
       - registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.17-openshift-4.11
   to: prometheus-config-reloader
+- dockerfile_path: Dockerfile.admission-webhook.ocp
+  from: base
+  inputs:
+    ocp_builder_rhel-8-golang-1.17-openshift-4.11:
+      as:
+      - registry.ci.openshift.org/ocp/builder:rhel-8-golang-1.17-openshift-4.11
+  to: prometheus-admission-webhook
```
Comment (fpetkovski) [inline ci-operator/config/openshift/prometheus-operator/openshift-prometheus-operator-master.yaml:34]: Nit: 
```suggestion
  to: prometheus-operator-admission-webhook
```

**Comment by simonpasquier** [comment]
/assign @bparees

In upstream Prometheus operator, the admission webhook is now a standalone binary separated from the operator itself so we need a new container image for it. @PhilipGough is also taking care of creating the downstream repository and image.

File: core-services/image-mirroring/openshift/mapping_origin_4_11 [hunk 2]
```diff
@@ -248,6 +248,7 @@ registry.ci.openshift.org/origin/4.11:prometheus-alertmanager quay.io/openshift/
 registry.ci.openshift.org/origin/4.11:prometheus-config-reloader quay.io/openshift/origin-prometheus-config-reloader:4.11 quay.io/openshift/origin-prometheus-config-reloader:4.11.0 quay.io/openshift/origin-prometheus-config-reloader:latest
 registry.ci.openshift.org/origin/4.11:prometheus-node-exporter quay.io/openshift/origin-prometheus-node-exporter:4.11 quay.io/openshift/origin-prometheus-node-exporter:4.11.0 quay.io/openshift/origin-prometheus-node-exporter:latest
 registry.ci.openshift.org/origin/4.11:prometheus-operator quay.io/openshift/origin-prometheus-operator:4.11 quay.io/openshift/origin-prometheus-operator:4.11.0 quay.io/openshift/origin-prometheus-operator:latest
+registry.ci.openshift.org/origin/4.11:prometheus-operator-admission-webhook quay.io/openshift/origin-prometheus-operator-admission-webhook:4.11 quay.io/openshift/origin-prometheus-operator-admission-webhook:4.11.0 quay.io/openshift/origin-prometheus-operator-admission-webhook:latest
```
Comment (bparees) [inline core-services/image-mirroring/openshift/mapping_origin_4_11:251]: name looks ok, assuming we decide it really makes sense to create a separate image for this binary vs embedding it in an existing image.

**Comment by PhilipGough** [comment]
@bparees the thinking was that we would like to remain consistent with upstream and to re-use artefacts like jsonnet etc

**Pull request merged**